ProcessLib/HydroMechanics/HydroMechanicsProcess-impl.h

```diff
@@ -188,43 +265,113 @@ void HydroMechanicsProcess<DisplacementDim>::assembleConcreteProcess(
 
 template <int DisplacementDim>
 void HydroMechanicsProcess<DisplacementDim>::
-    assembleWithJacobianConcreteProcess(
-        const double t, GlobalVector const& x, GlobalVector const& xdot,
-        const double dxdot_dx, const double dx_dx, GlobalMatrix& M,
-        GlobalMatrix& K, GlobalVector& b, GlobalMatrix& Jac)
+    assembleWithJacobianConcreteProcess(const double t, GlobalVector const& x,
+                                        GlobalVector const& xdot,
+                                        const double dxdot_dx,
+                                        const double dx_dx, GlobalMatrix& M,
+                                        GlobalMatrix& K, GlobalVector& b,
+                                        GlobalMatrix& Jac)
 {
-    DBUG("AssembleJacobian HydroMechanicsProcess.");
+    // For the monolithic scheme
+    if (_use_monolithic_scheme)
+    {
+        DBUG(
+            "Assemble the Jacobian of HydroMechanics for the monolithic"
+            " scheme.");
+        // Call global assembler for each local assembly item.
+        GlobalExecutor::executeMemberDereferenced(
+            _global_assembler, &VectorMatrixAssembler::assembleWithJacobian,
+            _local_assemblers, *_local_to_global_index_map, t, x, xdot,
+            dxdot_dx, dx_dx, M, K, b, Jac, _coupled_solutions, nullptr);
+        return;
+    }
+
+    // For the staggered scheme
+    if (_coupled_solutions->process_id == 0)
+    {
+        DBUG(
+            "Assemble the Jacobian equations of liquid fluid process in "
+            "HydroMechanics for the staggered scheme.");
+    }
+    else
+    {
+        DBUG(
+            "Assemble the Jacobian equations of mechanical process in "
+            "HydroMechanics for the staggered scheme.");
+    }
 
-    // Call global assembler for each local assembly item.
     GlobalExecutor::executeMemberDereferenced(
         _global_assembler, &VectorMatrixAssembler::assembleWithJacobian,
         _local_assemblers, *_local_to_global_index_map, t, x, xdot, dxdot_dx,
-        dx_dx, M, K, b, Jac, _coupled_solutions);
+        dx_dx, M, K, b, Jac, _coupled_solutions,
+        _local_to_global_index_map_with_base_nodes.get());
 }
+
 template <int DisplacementDim>
 void HydroMechanicsProcess<DisplacementDim>::preTimestepConcreteProcess(
     GlobalVector const& x, double const t, double const dt,
-    const int /*process_id*/)
+    const int process_id)
 {
     DBUG("PreTimestep HydroMechanicsProcess.");
 
     _process_data.dt = dt;
     _process_data.t = t;
 
-    GlobalExecutor::executeMemberOnDereferenced(
-        &LocalAssemblerInterface::preTimestep, _local_assemblers,
-        *_local_to_global_index_map, x, t, dt);
+    // If monolithic scheme is used or the equation of deformation is solved in
+    // the staggered scheme.
```

Comment:
For understanding: How is it guaranteed, that `preTimestep` is called first in the case of staggered scheme before the other (staggered) process is executed?

Reply:
If is called before a time step. For HydroMechanicsProcess, the function only updates strain, stress.